File: src/leetcode/python/two_pointers/ContainerWithMostWater_11.py
class ContainerWithMostWater_11(object):
    def maxArea(self, height):
        head = 0
        tail = len(height) - 1
        area = 0
        high = 0
        while head < tail:
            high = min(height[head], height[tail])
            area = max(area, (tail - head) * high)
            while height[head] <= high and head < tail:
                head += 1
            while height[tail] <= high and head < tail:
                tail -= 1
        return area

File: src/leetcode/java/medium/ContainerWithMostWater_11.java
package leetcode.java.medium;

public class ContainerWithMostWater_11 {
  public int maxArea(int[] height) {
    int area = 0;
    int left = 0, right = height.length - 1;
    int high;
    while (left < right) {
      high = Math.min(height[left], height[right]);
      area = Math.max(area, (right - left) * high);
      while (height[left] <= high && left < right) left++;
      while (height[right] <= high && left < right) right--;
    }
    return area;
  }
}
